server/src/main/java/com/epam/aidial/core/server/service/ApplicationService.java

```diff
@@ -131,9 +133,11 @@ public List<Application> getSharedApplications(ProxyContext context) {
             ResourceDescriptor resource = ResourceDescriptorFactory.fromAnyUrl(meta.getUrl(), encryptionService);
 
             if (meta instanceof ResourceItemMetadata) {
-                list.add(getApplication(resource).getValue());
+                Application application = getApplication(resource, context).getValue();
+                application = CustomApplicationUtils.filterCustomClientPropertiesWhenNoWriteAccess(context, resource, application);
```

Comment:
Why does getApplication() not filter them?

Reply:
![image](https://github.com/user-attachments/assets/beb2356c-15bd-4fa7-8790-e10edda65df3)
here is the one usage of getSharedApplications - so technically there is not need to move filtering outside the getSharedApplications method just because getSharedApplications is also a member of ApplicationService.